pkg/providers/heroku/applications.go

```diff
@@ -0,0 +1,42 @@
+package heroku
+
+import (
+	"context"
+
+	heroku "github.com/heroku/heroku-go/v5"
+	"github.com/projectdiscovery/cloudlist/pkg/schema"
+)
+
+// instanceProvider is an instance provider for Heroku API
+type instanceProvider struct {
+	profile string
+	client  *heroku.Service
+}
+
+// GetResource returns all the applications for the Heroku provider.
+func (d *instanceProvider) GetResource(ctx context.Context) (*schema.Resources, error) {
+	list := &schema.Resources{}
+
+	apps, err := d.client.AppList(context.TODO(), &heroku.ListRange{Field: "name"})
```

Comment:
We should see how the heroku API handles pagination, and also offer support for paginating the responses. As some heroku accounts have many apps, it'd be wise to do so.

Reply:
We can't support pagination for now because of the `heroku-sdk` limitations. Although I have changed `Max` value in `ListRange` so that it handles apps upto `1000`.